src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -356,6 +356,7 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                 Console.Error.WriteLine($"[ERROR] {ex.ToString()}");
                 collectionStopped = true;
                 ret = ReturnCode.TracingError;
+                processStarted = !ex.Message.StartsWith("Failed to start");
```

Comment:
```suggestion
                processStarted = !(ex is InvalidOperationException && ex.Message.StartsWith("Failed to start"));
```

Hmm. Not sure if we want to depend on this string never changing. This might be a little more robust. The other case where we could be a bit more resilient is using a bit more of a specific exception in 

https://github.com/dotnet/diagnostics/blob/1bdd90c6f4a1a4a1800aae57bd195026f36645e7/src/Tools/Common/ReversedServerHelpers/ReversedServerHelpers.cs#L219.

That way we could depend on some metadata on the exception for this.

Reply:
Taking a control flow dependency on an exception message (or on nearly any string literal) should be avoided because these strings could change. Most developers will expect that if text looks like it is intended for humans to read then they are free to change that text at any time if it would help humans understand the meaning better. For error handling control flow the conventional pattern is to make decisions based on the type of Exception object, or occasionally on other non-string exception properties. For example you might see:

```C#
try
{
   File.Read(...)
}
catch(FileNotFoundException e)
{
    Console.WriteLine("The file you requested didn't exist");
}
catch(IOException e)
{
    Console.WriteLine("The file you requested couldn't be read");
}
```

If we remove the processStarted field as I suggested above this is probably moot, but maybe useful in the future.